### PR TITLE
Reorganise new multilevel facets

### PR DESCRIPTION
**Pull request by sanjithkumar017**

### CODE REVIEW

File: src/modules/combinedFacets/generateCombinedFacets/GenerateCombinedFacets.js [hunk 1]
```diff
@@ -296,14 +232,17 @@ class GenerateCombinedFacets extends React.Component {
...
+                                                data-testid="UNX_searchFacets"
                                             />
                                         </div>
                                     )}
                                     <List
                                         items={filteredValues}
-                                        ListItem={multilevelFacetItemComponent}
-                                        idAttribute={'name'}
+                                        ListItem={
+                                            multilevelFacetItemComponent ||
```
[the remark this answers is not in the document]
Comment (sanjithkumar017) [reply src/modules/combinedFacets/generateCombinedFacets/GenerateCombinedFacets.js:242]: It is an instance.

File: src/common/facetUtils.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const facetName = event.target.dataset['unx_name'];
+    this.setState((existingState) => {
+        const updatedFacets = existingState[stateKey].map((facetValue) => {
+            if (facetValue.facetName === facetName) {
+                return {
+                    ...facetValue,
+                    viewLess: !facetValue.viewLess
+                };
+            }
+            return { ...facetValue };
```
Comment (Anupama-Unbxd) [inline src/common/facetUtils.js:11]: if `facetValue` is an object, we can just do `return facetvalue` ?

File: src/common/facetUtils.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const facetName = event.target.dataset['unx_name'];
+    this.setState((existingState) => {
+        const updatedFacets = existingState[stateKey].map((facetValue) => {
+            if (facetValue.facetName === facetName) {
+                return {
+                    ...facetValue,
+                    isOpen: !facetValue.isOpen
+                };
+            }
+            return { ...facetValue };
```
Comment (Anupama-Unbxd) [inline src/common/facetUtils.js:31]: same as above

File: src/common/facetUtils.js (hunk 2 above)
Comment (sanjithkumar017) [reply src/common/facetUtils.js:11]: Shouldn't we be passing new data. functional programming and all? So react detects a change.

**Pull request merged**